src/Container/Container.php

```diff
@@ -165,4 +222,44 @@ public function __call($name, $arguments)
             throw new \BadMethodCallException("No application method named {$name}()");
         }
     }
+
+    /**
+     * Instantiate an object of named class using reflection to recursively to resolve dependencies
+     *
+     * @param string $className
+     * @return mixed
+     * @throws \ReflectionException
+     */
+    protected function resolveInstance(string $className) {
+        $class = new \ReflectionClass($className);
+
+        if (!$class->isInstantiable()) {
+            throw new \ReflectionException("$className cannot be instantiated");
+        }
+
+        return $class->newInstanceArgs($this->resolveParameters($class->getConstructor()->getParameters()));
```

Comment:
this is a long line with three levels of function wrapping and not easy to read. Better make this more readable.